tools/pot/openvino/tools/pot/data_loaders/syntetic_image_loader.py

```diff
@@ -0,0 +1,293 @@
+        return image
+
+    def make_patch3_3(self, mask, patch_color):
+        patch_color = np.array(patch_color)
+        patch = np.zeros((3, 3, 3), np.uint8)
+        for i in range(3):
+            for j in range(3):
+                patch[i ,j, :] = patch_color * int(mask[i*3+j])
+        return patch
+
+
+class SynteticImageLoader(ImageLoader):
+    def __init__(self, config):
+        np.random.seed(seed=42)
+        self._cpu_count = os.cpu_count()
+        self._weights = [
+            0.2, 1, 1, 1, 1, 1,
+            0.6, 1, 1, 1, 1, 1,
+            1, 1, 1, 1, 1, 1,
+            1.4, 1, 1, 1, 1, 1,
+            1.8, 1, 1, 1, 1, 1,
+            1, 0.2, 1, 1, 1,1,
+            1, 0.6, 1, 1, 1, 1,
+            1, 1.4, 1, 1, 1, 1,
+            1, 1.8, 1, 1, 1, 1,
+            1, 1, 0.2, 1, 1, 1,
+            1, 1, 0.6, 1, 1, 1,
+            1, 1, 1.4, 1, 1, 1,
+            1, 1, 1.8, 1, 1, 1,
+            1, 1, 1, 0.2, 1, 1,
+            1, 1, 1, 0.6, 1, 1,
+            1, 1, 1, 1.4, 1, 1,
+            1, 1, 1, 1.8, 1, 1,
+            1, 1, 1, 1, 0.2, 1,
+            1, 1, 1, 1, 0.6, 1,
+            1, 1, 1, 1, 1.4, 1,
+            1, 1, 1, 1, 1.8, 1,
+            1, 1, 1, 1, 1, 0.2,
+            1, 1, 1, 1, 1, 0.6,
+            1, 1, 1, 1, 1, 1.4,
+            1, 1, 1, 1, 1, 1.8
+        ]
+
+        if not self.shape:
+            inputs = get_nodes_by_type(self.model, ['Parameter'], recursively=False)
+            if len(inputs) > 1:
+                raise RuntimeError('SynteticImageLoader supports networks with single image input'
+                                   'Actual inputs number: {}'.format(len(inputs)))
+            self.shape = inputs[0].shape
+
+        # how to check dynamic shape like min...max?
+        assert len(self.shape) == 3 or len(self.shape) == 4 and self.shape[-1] != -1 and self.shape[-2] != -1
+
+        self._net = cv.dnn.readNetFromCaffe('colorization_deploy_v2.prototxt', 'colorization_release_v2.caffemodel')
+        pts_in_hull = np.load('pts_in_hull.npy').transpose().reshape(2, 313, 1, 1)
+        self._net.getLayer(self._net.getLayerId('class8_ab')).blobs = [pts_in_hull.astype(np.float32)]
+        self._net.getLayer(self._net.getLayerId('conv8_313_rh')).blobs = [np.full([1, 313], 2.606, np.float32)]
+
+        if not os.path.exists(self.dataset_dir):
+            logger.info(f'Syntetic dataset will be stored in {config.dataset_dir}')
+            os.mkdir(self.dataset_dir)
+
+            if config.generate_data:
+                # progress bar?
+                self.generate_dataset()
+            else:
+                # self.download_images()
+                raise NotImplementedError
+        else:
+            logger.info(f'Dataset was found in {config.dataset_dir}')
+
+        super().__init__(config)
+
+    def download_images(self):
+        URL = ''
+        response  = requests.get(URL, stream=True)
+        archive_path = os.path.join(self.dataset_dir, 'FractalDB-1k.tar.gz')
```

Comment:
I would recommend naming it as `synthetic_images.tag.gz` and avoiding a connection to FractalDB or whatever. At some point, we can switch to another type of image.